subsys/bluetooth/services/fast_pair/fp_bond_manager.c

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <errno.h>
+#include <zephyr/bluetooth/conn.h>
+
+#include <bluetooth/services/fast_pair/fast_pair.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(fast_pair, CONFIG_BT_FAST_PAIR_LOG_LEVEL);
+
+#include "fp_activation.h"
+#include "fp_storage_ak_bond.h"
+
+struct bond_find_ctx {
+	bool bond_found;
+	const bt_addr_le_t *bond_addr;
+};
+
+static bool is_enabled;
+
+static void identity_resolved(struct bt_conn *conn, const bt_addr_le_t *rpa,
+			      const bt_addr_le_t *identity)
+{
+	if (!bt_fast_pair_is_ready()) {
+		return;
+	}
+
+	LOG_DBG("Identity resolved");
+
+	fp_storage_ak_bond_conn_addr_update(conn, identity);
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	if (!bt_fast_pair_is_ready()) {
+		return;
+	}
+
+	fp_storage_ak_bond_conn_finalize(conn);
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.disconnected = disconnected,
+	.identity_resolved = identity_resolved,
+};
+
+static void bond_deleted(uint8_t id, const bt_addr_le_t *peer)
+{
+	fp_storage_ak_bond_delete(peer);
+}
+
+static struct bt_conn_auth_info_cb conn_auth_info_callbacks = {
+	.bond_deleted = bond_deleted,
+};
+
+static void bond_address_cmp(const struct bt_bond_info *info, void *user_data)
+{
+	struct bond_find_ctx *ctx = user_data;
+
+	if (bt_addr_le_eq(ctx->bond_addr, &info->addr)) {
+		__ASSERT_NO_MSG(!ctx->bond_found);
+		ctx->bond_found = true;
+	}
+}
+
+static int bond_identity_find(uint8_t *identity, const bt_addr_le_t *addr)
+{
+	struct bond_find_ctx ctx = {
+		.bond_found = false,
+		.bond_addr = addr,
+	};
+
+	for (uint8_t id = 0; id < CONFIG_BT_ID_MAX; id++) {
+		bt_foreach_bond(id, bond_address_cmp, &ctx);
+		if (ctx.bond_found) {
+			if (identity) {
+				*identity = id;
+			}
+		}
+	}
+
+	return ctx.bond_found ? 0 : -ESRCH;
+}
+
+static int bond_remove(const bt_addr_le_t *addr)
+{
+	__ASSERT_NO_MSG(is_enabled);
```

Comment:
Would fail if Fast Pair factory reset is called while Fast Pair subsystem is disabled.